modules/apps/object/object-test/src/testFunctional/tests/CreateObject.testcase

```diff
@@ -1759,6 +1759,67 @@ definition {
 		ObjectPortlet.assertDeleteEntryButtonNotPresent(entry = "Custom Object 42");
 	}
 
+	@description = "LPS-151877 - Verify that after the entry of an object definition with account restriction is saved, the account field can not be edited"
+	@priority = 4
+	test CannotEditEntryWithAccountRestriction {
+		task ("Given: Two accounts, a Custom Object and Relationship Account to Custom Object are created (Via API)") {
+			for (var accountNum : list "1,2") {
+				JSONAccountEntry.addAccountEntry(
+					accountEntryName = "Account Test ${accountNum}",
+					accountEntryType = "Business");
+			}
+
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 176837",
+				objectName = "CustomObject176837",
+				pluralLabelName = "Custom Objects 176837");
+
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "AccountEntry",
+				objectName_2 = "CustomObject176837",
+				relationshipLabel = "RelationshipAccount",
+				relationshipName = "relationshipAccount",
+				relationshipType = "oneToMany");
+		}
+
+		task ("And given: A layout is created and Account Restriction is enabled") {
+			ObjectAdmin.addObjectLayoutViaAPI(
+				layoutName = "Layout Name",
+				objectName = "CustomObject176837");
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object");
+
+			CreateObject.activeToogleAccountRestriction();
+
+			CreateObject.selectAccountRestrictedField(accountRestrictedField = "RelationshipAccount");
+
+			Button.clickPublish();
+		}
+
+		task ("When: An entry is added") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject176837");
+
+			LexiconEntry.gotoAdd();
+
+			ObjectPortlet.chooseEntryOnRelationshipField(entryOption = "Account Test 1");
+
+			Button.clickSave();
+		}
+
+		task ("Then: The entry is read-only and is not possible to edit to Account 2") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject176837");
+
+			ObjectPortlet.viewEntryDetails(entry = "Account Test 1");
+
+			AssertClick(
+				key_entry = "Account Test 1",
+				locator1 = "ObjectPortlet#RELATIONSHIP_ENTRY_READ_ONLY",
+				value1 = ${key_entry});
```

Comment:
Using the variable here might make the code less readable. Even though can look a little repetitive, using the "Account Test 1" in both makes it faster for the reader to understand what's being accomplished.